unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs

```diff
@@ -66,6 +72,39 @@ public void Populate(Model chatEntryModel)
 
         LayoutRebuilder.ForceRebuildLayoutImmediate(body.transform as RectTransform);
         LayoutRebuilder.ForceRebuildLayoutImmediate(username.transform as RectTransform);
+
+        if (this.enabled)
+            group.alpha = 0;
+    }
+
+    [SerializeField] CanvasGroup group;
+
+    public void SetFadeout(bool enabled)
+    {
+        if (!enabled)
+        {
+            group.alpha = 1;
+            this.enabled = false;
+            return;
+        }
+
+        this.enabled = true;
+    }
+
+    private void Update()
+    {
+        double fadeTime = (double)(model.timestamp / 1000.0) + timeToFade;
```

Comment:
I'd love to have an escape line here to avoid calculating the time stuff if the entry is fully visible, etc. but not sure we can do it.

Reply:
> I'd love to have an escape line here to avoid calculating the time stuff if the entry is fully visible, etc. but not sure we can do it.

When `SetFadeout` is set to disable it disable the component, so there are no calls to `Update`